sklearn/preprocessing/data.py

```diff
@@ -2873,3 +2873,37 @@ def inverse_transform(self, X):
                     X_tr[mask, idx] = None
 
         return X_tr
+
+    def get_feature_names(self, input_features=None):
+        """
+        Return feature names for output features
+
+        Parameters
+        ----------
+        input_features : list of string, length n_features, optional
+            String names for input features if available. By default,
+            "x0", "x1", ... "xn_features" is used.
+
+        Returns
+        -------
+        output_feature_names : list of string, length n_output_features
+
+        """
+        cats = self.categories_
+        feature_names = []
+        if input_features is None:
+            input_features = ['x%d' % i for i in range(len(cats))]
+
+        cats2 = []
+        for i in range(len(cats)):
+            temp = []
+            for t in cats[i]:
+                temp.append(str(t))
+            cats2.append(np.array(temp))
+
+        for i in range(len(cats)):
+            t = [input_features[i] + '_' + f for f in cats2[i]]
+            feature_names.append(t)
+
+        feature_names = np.concatenate(feature_names).tolist()
```

Comment:
thats a weird way to join lists: you converte them to numpy arrays, concatenate them, and convert them back to lists.

You could probably just do a single list comprehension for everything.